crates/sovereign/sovereign-chain-components/src/sovereign/impls/sovereign_to_cosmos/client/update_client_payload.rs

```diff
@@ -126,10 +131,22 @@ where
             .await
             .map_err(|e| eyre!("Error creating headers from DA chain: {e:?}"))?;
 
+        let chain_status = chain.query_chain_status_at_height(target_height).await?;
```

Comment:
I am not sure if we have to have a generalized version of `query_chain_status_at_height`. It seems that the use is specific to Sovereign. So perhaps we can do the JSON RPC query directly.

Reply:
The way I see, cosmos chain doesn't need it, becuase we call `build_header` from hermes.

This step is basically `build_header` for sovereign. Considering this, I will let you make the decision.

Reply:
I moved the trait to `sovereign-rollup-components`, since it is sovereign-specific.